test/document_stores/test_faiss.py

```diff
@@ -8,294 +9,266 @@
+        reloaded_document_store = FAISSDocumentStore.load(
+            tmp_path / "haystack_test_faiss", config_path=tmp_path / "custom_path.json"
+        )
 
-    retriever = EmbeddingRetriever(
-        document_store=document_store, embedding_model="deepset/sentence_bert", use_gpu=False
-    )
-    result = retriever.retrieve(query="How to test this?")
+        # check faiss index is restored
+        assert reloaded_document_store.faiss_indexes[ds.index].ntotal == len(documents_embedding_only)
+        # check if documents are restored
+        assert len(reloaded_document_store.get_all_documents()) == len(documents_embedding_only)
+        # Check if the init parameters are kept
+        assert not reloaded_document_store.progress_bar
 
-    assert len(result) == len(DOCUMENTS)
-    assert type(result[0]) == Document
+        # test loading the index via init
+        new_document_store = FAISSDocumentStore(
+            faiss_index_path=tmp_path / "haystack_test_faiss", faiss_config_path=tmp_path / "custom_path.json"
+        )
 
-    # Cleanup
-    document_store.faiss_indexes[document_store.index].reset()
+        # check faiss index is restored
+        assert new_document_store.faiss_indexes[ds.index].ntotal == len(documents_embedding_only)
+        # check if documents are restored
+        assert len(new_document_store.get_all_documents()) == len(documents_embedding_only)
+        # Check if the init parameters are kept
+        assert not new_document_store.progress_bar
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("index_buffer_size", [10_000, 2])
+    def test_write_index_docs(self, ds, documents_embedding_only, index_buffer_size):
+        batch_size = 2
+        ds.index_buffer_size = index_buffer_size
+
+        # Write in small batches
+        for i in range(0, len(documents_embedding_only), batch_size):
+            ds.write_documents(documents_embedding_only[i : i + batch_size])
+
+        documents_indexed = ds.get_all_documents()
+        assert len(documents_indexed) == len(documents_embedding_only)
+
+        # test if correct vectors are associated with docs
+        for i, doc in enumerate(documents_indexed):
+            # we currently don't get the embeddings back when we call document_store.get_all_documents()
+            original_doc = [d for d in documents_embedding_only if d.content == doc.content][0]
+            stored_emb = ds.faiss_indexes[ds.index].reconstruct(int(doc.meta["vector_id"]))
+            # compare original input vec with stored one (ignore extra dim added by hnsw)
+            # original input vec is normalized as faiss only stores normalized vectors
+            assert np.allclose(original_doc.embedding / np.linalg.norm(original_doc.embedding), stored_emb, rtol=0.01)
+
+    @pytest.mark.integration
+    def test_write_docs_different_indexes(self, ds, documents_embedding_only):
```

Comment:
I think it would be better to write different documents in the two indices to make sure `write_documents` works as expected (in this test and in the following one)